src/kemal.cr

```diff
@@ -89,6 +89,8 @@ module Kemal
   end
 
   private def self.setup_trap_signal
+    return if Kemal.config.disable_trap_signal
```

Comment:
I would suggest to move this guard into `Kemal.run`. There is already a similar condition based on config environment.